### DISCUSSION TITLE
intital connection to a very big file takes a lot of time

### DISCUSSION THREAD

**Discussion by djouallah** [duckdb/duckdb#11768] [Q&A] [answered]
I have a 300 GB duckdb file, the performance is good when running arbitrary queries, but I notice the initial connection to  the file take really a long time, what is involved in reading the file, I thought it should be just a lightweight operation ?

**Comment by szarnyasg** [comment]
Hi @djouallah, thanks for this question. The initial connection step is definitely something that we can elaborate on in the documentation. In your use case, are you accessing the file on a local disk or a networked disk; and is it an SSD or NVMe disk?

**Reply by djouallah** [reply]
from an object store ( microsoft Fabric)

**Reply by Tishj** [reply]
Is there a big WAL file that needs to be replayed in those connections?

**Reply by djouallah** [reply]
no WAL

**Reply by szarnyasg** [reply, marked as answer]
Attaching to a remote DB is not fully optimized yet - we have a few long-hanging fruits to tackle to make this faster.

**Comment by djouallah** [comment]
Thanks, it will be nice to have that initial connection fixed, anyway so far the whole thing is promising (  Query 7,10,13 and 18 timeout) 
![image](https://github.com/duckdb/duckdb/assets/12554469/b6947bb7-bfe2-4021-bb60-ce8a9da8c7bc)
